### PR TITLE
new collapse button

### PR DESCRIPTION
**Pull request by hughockey**

### CODE REVIEW

File: src/Index/Index.ts [hunk 1]
```diff
@@ -13,18 +13,27 @@ const template = html<IndexComponent>`<div class="heading">
...
+
 export default class IndexComponent extends FASTElement {
   @attr violationCount = 0;
+  @attr({mode: 'boolean'}) collapsed: boolean = JSON.parse(sessionStorage.getItem('collapsed'));
+
+  collapse() {
+    this.collapsed = !this.collapsed;
+    sessionStorage.setItem('collapsed', JSON.stringify(this.collapsed)); 
+  }
+
```
Comment (m4thieulavoie) [inline src/Index/Index.ts:38]: [Nit] Useless empty line

File: src/Index/Index.ts [hunk 2]
```diff
@@ -13,18 +13,27 @@ const template = html<IndexComponent>`<div class="heading">
...
   <hr />
-  <slot></slot>`;
+  <slot class="${x => x.collapsed ? 'hide':''}"></slot>`;
 
 @customElement({
   name: "abr-index",
   template,
   styles,
 })
+
```
Comment (m4thieulavoie) [inline src/Index/Index.ts:29]: Why adding an empty line here? I think decorators should stick to their class statement

File: src/Index/Index.scss [hunk 3]
```diff
@@ -15,6 +15,10 @@
   border-radius: 6px;
 }
 
+.hide {
```
Comment (m4thieulavoie) [inline src/Index/Index.scss:18]: There's no need for a class here, you can do something like this, which is quite simpler
```suggestion
:host([collapsed]) slot {
```

File: src/Index/Index.ts [hunk 4]
```diff
@@ -13,18 +13,27 @@ const template = html<IndexComponent>`<div class="heading">
...
 export default class IndexComponent extends FASTElement {
   @attr violationCount = 0;
+  @attr({mode: 'boolean'}) collapsed: boolean = JSON.parse(sessionStorage.getItem('collapsed'));
```
Comment (m4thieulavoie) [inline src/Index/Index.ts:32]: We could take advantage of typescript here
```suggestion
  @attr({mode: 'boolean'}) collapsed = !!sessionStorage.getItem('collapsed');
```

File: src/Index/Index.ts [hunk 5]
```diff
@@ -13,18 +13,27 @@ const template = html<IndexComponent>`<div class="heading">
       this web page
     </p>
     <div class="buttons">
-      <button class="refresh" @click=${() => triggerAxeCore()}>Refresh</button
-      ><button class="close" @click=${(x) => x.remove()}>Close</button>
+      <button class="refresh" @click=${() => triggerAxeCore()}>Refresh</button>
+      <button @click=${(x) => x.collapse()}>Collapse</button>
```
Comment (m4thieulavoie) [inline src/Index/Index.ts:17]: We should update the button text when the component is collapsed

**Review by m4thieulavoie — CHANGES_REQUESTED** [review]

File: src/Index/Index.ts (hunk 4 above)
Comment (hughockey) [reply src/Index/Index.ts:32]: I put your suggestion in my code and like I said to you earlier, I have an issue with the !! . Even if the value in the sessionStorage is false, with the coalesce the value is always true.

I don't really know what we are doing different but I literally copied the line and it doesn't work, What do we do?

Comment (m4thieulavoie) [reply src/Index/Index.ts:32]: You're right, let's leave it this way for now :+1:

File: src/Index/Index.ts [hunk 6]
```diff
@@ -13,12 +13,13 @@ const template = html<IndexComponent>`<div class="heading">
...
-      <button class="refresh" @click=${() => triggerAxeCore()}>Refresh</button
-      ><button class="close" @click=${(x) => x.remove()}>Close</button>
+      <button class="refresh" @click=${() => triggerAxeCore()}>Refresh</button>
+      <button @click=${(x) => x.collapse()}>${(x) => x.collapsed ? 'Expand' : 'Collapse'}</button>
+      <button class="close" @click=${(x) => x.remove()}>Close</button>
     </div>
   </div>
   <hr />
-  <slot></slot>`;
+  <slot class="${x => x.collapsed}"></slot>`;
```
Comment (m4thieulavoie) [inline src/Index/Index.ts:22]: This change is not required anymore

**Review by m4thieulavoie — CHANGES_REQUESTED** [review]

File: src/Index/Index.ts (hunk 6 above)
Comment (hughockey) [reply src/Index/Index.ts]: Yeah, sorry, forgot that part.

**Review by m4thieulavoie — APPROVED** [review]

**Pull request merged**